web-client/src/presenter/actions/hasTrialSessionLocationChangedAction.test.ts

```diff
@@ -0,0 +1,253 @@
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.remote,
+        },
+      },
+    });
+
+    expect(unchangedMock.mock.calls.length).toEqual(1);
+    expect(updatedMock.mock.calls.length).toEqual(0);
+  });
+
+  it('should call "unchanged" path when either location data is not calendared', async () => {
+    applicationContext
+      .getUseCases()
+      .getTrialSessionOpenCasesCountInteractor.mockReturnValue({
+        calendaredCaseEntitiesCount: 1,
+        casesThatShouldReceiveNoticesCount: 1,
+      });
+
+    const CURRENT_LOCATION: TrialSessionLocationInfo = {
+      address1: 'TEST_ADDRESS_1',
+      address2: 'TEST_ADDTESS_2',
+      city: 'TEST_CITY',
+      courthouseName: 'TEST_COURTHOUSE_NAME',
+      postalCode: 'TEST_POSTAL_CODE',
+      proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+      state: 'TEST_STATE',
+      trialLocation: 'TEST_TRIAL_LOCATION',
+    };
+
+    const UPDATED_LOCATION: TrialSessionLocationInfo = {
+      address1: 'TEST_ADDRESS_1',
+      address2: 'TEST_ADDTESS_2',
+      city: 'TEST_CITY',
+      courthouseName: 'TEST_COURTHOUSE_NAME',
+      postalCode: 'TEST_POSTAL_CODE',
+      proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+      state: 'TEST_STATE',
+      trialLocation: 'UPDATED__TEST_TRIAL_LOCATION',
+    };
+
+    await runAction(hasTrialSessionLocationChangedAction, {
+      modules: {
+        presenter,
+      },
+      state: {
+        form: {
+          ...UPDATED_LOCATION,
+          isCalendared: false,
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+        },
+        formattedTrialSessionDetails: {
+          ...CURRENT_LOCATION,
+          isCalendared: false,
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+        },
+      },
+    });
+
+    expect(unchangedMock.mock.calls.length).toEqual(1);
+    expect(updatedMock.mock.calls.length).toEqual(0);
+  });
+
+  it('should call "unchanged" path when location information is the same', async () => {
+    applicationContext
+      .getUseCases()
+      .getTrialSessionOpenCasesCountInteractor.mockReturnValue({
+        calendaredCaseEntitiesCount: 1,
+        casesThatShouldReceiveNoticesCount: 1,
+      });
+
+    await runAction(hasTrialSessionLocationChangedAction, {
+      modules: {
+        presenter,
+      },
+      state: {
+        form: {
+          address1: 'TEST_ADDRESS_1',
+          isCalendared: true,
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+        },
+        formattedTrialSessionDetails: {
+          address1: 'TEST_ADDRESS_1',
+          isCalendared: true,
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+        },
+      },
+    });
+
+    expect(unchangedMock.mock.calls.length).toEqual(1);
+    expect(updatedMock.mock.calls.length).toEqual(0);
+  });
+
+  it('should call "unchanged" path when location information has been updated but there are no cases to generate notices', async () => {
```

Comment:
I'm a little nervous about the naming of `hasTrialSessionLocationChangedAction` if it is checking more than whether or not the trial session location has changed. I wonder if something like `needsToGenerateNoticeOfChangedTrialLocationAction` would be better.